subsys/secure_storage/src/aead/secure_backend_aead.c

```diff
@@ -0,0 +1,338 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/sys/util.h>
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(internal_secure_aead, CONFIG_SECURE_STORAGE_LOG_LEVEL);
+
+#include <string.h>
+
+#include "../secure_storage_backend.h"
+#include "../storage_backend.h"
+#include "aead_key.h"
+#include "aead_nonce.h"
+#include "aead_crypt.h"
+
+/* Common data and metadata suffixes. */
+#define SECURE_STORAGE_AEAD_BACKEND_FILENAME_SUFFIX_SIZE  ".size"
+#define SECURE_STORAGE_AEAD_BACKEND_FILENAME_SUFFIX_FLAGS ".flags"
+#define SECURE_STORAGE_AEAD_BACKEND_FILENAME_SUFFIX_DATA  ".data"
+#define SECURE_STORAGE_AEAD_BACKEND_FILENAME_SUFFIX_NONCE ".nonce"
+
+#define SECURE_STORAGE_MAX_ASSET_SIZE CONFIG_SECURE_STORAGE_AEAD_BACKEND_MAX_DATA_SIZE
+
+/*
+ * AEAD based Authenticated Encrypted trust implementation
+ *
+ * Actual implementation uses:
+ * - Hexadecimal UID imported as KEY
+ * - UID+Flags+Size as additional parameter
+ * - Nonce is a number that is incremented for each encryption.
+ * - Tag is left at the end of output data
+ */
+
+#define AEAD_TAG_SIZE	16
+#define AEAD_NONCE_SIZE 12
+
+/* Max storage size for the encrypted or decrypted output */
+#define AEAD_MAX_BUF_SIZE ROUND_UP(SECURE_STORAGE_MAX_ASSET_SIZE + AEAD_TAG_SIZE, AEAD_TAG_SIZE)
+
+/* Additional data structure */
+struct aead_additional_data {
+	psa_storage_uid_t uid;
+	psa_storage_create_flags_t flags;
+	size_t size;
+};
+
+/* Temporary AEAD encryption/decryption buffers */
+static uint8_t aead_buf[AEAD_MAX_BUF_SIZE];
+static uint8_t data_buf[SECURE_STORAGE_MAX_ASSET_SIZE];
+
+static void safe_memset(void *dest, const size_t dest_size, const uint8_t ch, const size_t n)
```

Comment:
I think we should use mbedtls_platform_zeroise instead

Reply:
I could not find that function, can you tell me what to include to get that function

Reply:
It's part of mbedtls/platform_util.h, I think you you should have access to it.

Reply:
Ahh ok typo explains why I couldn't find it.. changed it